community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/ast/convert/commands/StatementConverters.scala

```diff
@@ -231,13 +231,15 @@ object StatementConverters {
   }
 
   implicit class RonjaHintConverter(val item: ast.UsingHint) extends AnyVal {
-    def asCommandStartHint(notifications: InternalNotificationLogger) = item match {
+    def asCommandStartHint(notifications: InternalNotificationLogger, plannerName: String) = item match {
       case ast.UsingIndexHint(identifier, label, property) =>
         Some(commands.SchemaIndex(identifier.name, label.name, property.name, commands.AnyIndex, None))
       case ast.UsingScanHint(identifier, label) =>
         Some(commands.NodeByLabel(identifier.name, label.name))
       case ast.UsingJoinHint(identifiers) =>
-        notifications.log(JoinHintUnsupportedNotification(identifiers.map(_.name).toSeq))
+        if (PlannerName(plannerName) == RulePlannerName) {
```

Comment:
It seems like this method is actually satisfying the index hints (creating commands for index hints), but not doing so for join hints. This 'hints' at the fact that join hints are solved somewhere else. If perhaps they were solved before this method and only unsatisfied hints passed in, then this method would have worked as is. Are we sure that they can only be satisfied later?